pkg/descheduler/strategies/rescheduler_mode.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package strategies
+
+import (
+	"github.com/golang/glog"
+
+	"github.com/kubernetes-incubator/descheduler/cmd/descheduler/app/options"
+	"github.com/kubernetes-incubator/descheduler/pkg/api"
+	"github.com/kubernetes-incubator/descheduler/pkg/descheduler/evictions"
+	podutil "github.com/kubernetes-incubator/descheduler/pkg/descheduler/pod"
+	"k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	priorityutil "k8s.io/kubernetes/plugin/pkg/scheduler/algorithm/priorities/util"
+)
+
+const (
+	criticalAddonsOnlyTaintKey = "CriticalAddonsOnly"
+)
+
+func RunInReschedulerMode(ds *options.DeschedulerServer, strategy api.DeschedulerStrategy, policyGroupVersion string, nodes []*v1.Node, nodepodCount nodePodEvictedCount) {
+	if !strategy.Enabled {
+		return
+	}
+	evictPodsOnlyInReschedulerMode(ds.Client, policyGroupVersion, nodes, ds.DryRun, nodepodCount, ds.MaxNoOfPodsToEvictPerNode)
+}
+
+func evictPodsOnlyInReschedulerMode(client clientset.Interface, policyGroupVersion string, nodes []*v1.Node, dryRun bool, nodepodCount nodePodEvictedCount, maxPodsToEvict int) int {
+	for _, node := range nodes {
+		criticalDSPods := getCriticalDaemonSetPodsOnNode(client, node)
+		if len(criticalDSPods) < 0 || !IsNodeUnderResourcePressure(client, node) {
+			glog.Infof("Node %v doesn't have critical pods or this node is not under resource pressure")
+			continue
+		}
+		// Add critical taint to node, so that default scheduler won't schedule pods onto this node, while we
+		// are making room for criticalDaemonSetPods.
+		if err := addTaintsToNode(client, node); err != nil {
+			glog.Fatalf("Error while applying taints. Let's skip this node.")
+			continue
+		}
+		for _, criticalDSPod := range criticalDSPods {
+			if err := makeRoomForCriticalDaemonSetPod(client, criticalDSPod, node, policyGroupVersion, dryRun); err != nil {
+				glog.Fatal("Error while making room for critical DS pods")
+				continue
+			}
+		}
+		// Release critical taint from node.
+		releaseTaintsFromNode(client, node)
```

Comment:
Note to myslf: Need to add number of pods evicted to ensure criticalDS pod runs smoothly.